### ISSUE TITLE
Possible client update?

### ISSUE THREAD

**Issue by rylanharper** [thetre97/gridsome-source-shopify#86]
Hey Travis,

Hope you have been well! I still use this plugin often when developing Shopify stores for clients and it always works wonderfully - thanks for all the effort put into this.

I am slowly moving over to Nuxt since it seems Gridsome hasn't been updated in a while (although its definitely not as easy to set up as Gridsome is), but in the meantime I was wondering if it would be possible to update the Api client to the latest `2022-7` version? There has been some parts added to the graphql api and it would be great to make use of them.

If not its totally cool. I get that this may no longer be in your "stack" since there hasn't been much Gridsome updates (although a small update did happen last week for Vue 2.7).

Also, if you haven't seen it, I updated that old Shopify/Sanity repo you helped me on a long time ago which is on my Github page!

**Comment by ziopod** [comment]
Also, some of deprecated API calls supports will be removed on April 1, 2023. 

From Shopify Partners : 

"
Breaking changes:

- Several previously deprecated fields have been removed from the Storefront API. Please check the release notes for a full list of changes. [View change.](https://shopify.dev/api/release-notes/2022-01#graphql-storefront-api-changes)
- Switch the Storefront API's HasMetafields.metafields from a paginated connection to a field that requires explicit identifiers for metafields to be returned. [View change.](https://shopify.dev/changelog/the-behavior-of-hasmetafields-metafields-has-changed)
- As of 2022-10, the Money scalar is being removed from the Storefront API, and fields with V2 ({field}V2: MoneyV2) suffixes are being deprecated. Please check the changelog notes for full details of the change. [View change.](https://shopify.dev/changelog/changes-to-graphql-api-moneyv2-related-fields)
"
Cheers

**Comment by rylanharper** [comment]
@ziopod I am confused on the metafield change... is `metafields` no longer viable to use on the product query?

**Comment by rylanharper** [comment]
@ziopod @thetre97 I made a fork of this repo and updated everything for the current `2023-01` API → [Here is the fork](https://github.com/rylanharper/gridsome-source-shopify). I have tested this out and it is working just fine... you can use it by running `yarn add https://github.com/rylanharper/gridsome-source-shopify`

Of course I am still a bit confused about the best approach for the `metafields` tags, I currently have the following where you would simply put in a `color` metafield on the product variants:
```
metafields(identifiers: { namespace: "custom", key: "color" }){
  key
  value
  namespace
}
```
Again, not sure if this repo if still supported, but I can make a pull request if so. Keep me updated. If not you can just use the fork I made:)

**Comment by travis-r6s** [comment]
@rylanharper As you need to specify specific identifiers (which can be an array of objects containing namespace + key, as above), I guess it would need a plugin config option where you specify the metafields you want to fetch?

**Comment by rylanharper** [comment]
It would, yes.. Haha I have no idea how to do that though unfortunately 😖 The fork I made just uses the current `2023-01` client with updated queries (articles, shop, product, etc.). I also changed the `transformedSrc` resolver to receive the new image `url` tag instead of the deprecated `originalSrc` tag. On top of this, I removed tags that are no longer used.

The `metafields` tag I made above (which is in the fork I made) just takes in a basic namespace & key I was using (in this case color), but you totally correct in that it would be an array of unique objects.

It seems like a lot of work to add that within the plugin config options though.. Its not an issue if that can't be added in (if you dont have the time). I am not really sure how many people still use this plugin to be honest since Gridsome is sort of suck in limbo.

I can submit a pull request with the changes I've made though if you want? Just let me know
